cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js

```diff
@@ -599,4 +600,85 @@ describe("Payment Method list using Constraint Graph flow tests", () => {
       });
     }
   );
+
+  context(
+    `
+    MCA1 -> Stripe configured with credit = { currency = "USD" }\n
+    MCA2 -> Novalnet configured with credit = { currency = "EUR" }\n
+    Payment is done with currency as as USD and no billing address\n
+    The resultant Payment Method list should only have credit with stripe\n
+    `,
+    () => {
+      before("seed global state", () => {
+        cy.task("getGlobalState").then((state) => {
+          globalState = new State(state);
+        });
+      });
+
+      it("merchant-create-call-test", () => {
+        cy.merchantCreateCallTest(fixtures.merchantCreateBody, globalState);
+      });
+
+      it("api-key-create-call-test", () => {
+        cy.apiKeyCreateTest(fixtures.apiKeyCreateBody, globalState);
+      });
+      it("customer-create-call-test", () => {
+        cy.createCustomerCallTest(fixtures.customerCreateBody, globalState);
+      });
+      // stripe connector create with card credit enabled in USD
+      it("connector-create-call-test", () => {
+        cy.createNamedConnectorCallTest(
+          "payment_processor",
+          fixtures.createConnectorBody,
+          cardCreditEnabledInUsd,
+          globalState,
+          "stripe",
+          "stripe_US_default"
+        );
+      });
+
+      // novalnet connector create with card credit enabled in EUR
+      it("connector-create-call-test", () => {
+        cy.createNamedConnectorCallTest(
+          "payment_processor",
+          fixtures.createConnectorBody,
+          cardCreditEnabledInEur,
+          globalState,
+          "novalnet",
+          "novalnet_DE_default"
+        );
+      });
+
+      // creating payment with currency as USD and no billing email
+      // billing.email is mandatory for novalnet
+      it("create-payment-call-test", () => {
+        const data = getConnectorDetails("stripe")["pm_list"]["PaymentIntent"];
+        const newData = {
+          ...data,
+          Request: data.RequestCurrencyUSD,
+          RequestCurrencyUSD: undefined, // we do not need this anymore
+        };
+
+        cy.createPaymentIntentTest(
+          createPaymentBodyWithCurrencyCountry("USD", "IN", "IN"),
+          newData,
+          "no_three_ds",
+          "automatic",
+          globalState
+        );
+      });
+
+      // payment method list should only have credit with stripe
+      it("payment-method-list-call-test", () => {
+        const data =
+          getConnectorDetails("stripe")["pm_list"]["PmListResponse"][
```

Comment:
if not asking too much, do you mind renaming the `Stripe.js` file to `Connector.js` since it is more meaningful and common across all the connectors from my unerstanding?

and put the content that you've added in `PaymentUtils/Novalnet.js` in this `Connector.js` file if non-existent.

Reply:
this is done!